plugin/cancel/README.md

```diff
@@ -0,0 +1,34 @@
+# cancel
+
+## Name
+
+*cancel* - a plugin that cancels a request's context after 5001 milliseconds.
+
+## Description
+
+The *cancel* plugin creates a canceling context for each request. It adds a timeout that gets
+triggered after 5001 milliseconds.
+
+The 5001 number is chosen because the default timeout for DNS clients is 5 seconds, after that they
```

Comment:
I know you like to avoid knobs, but I can see use cases for setting this even lower so making it customizable makes sense to me.